src/plugins/discover/public/application/components/discover_grid/get_render_cell_value.tsx

```diff
@@ -51,20 +52,84 @@ export const getRenderCellValueFn = (
     return <span>-</span>;
   }
 
+  if (
+    useNewFieldsApi &&
+    !field &&
+    row &&
+    row.fields &&
+    !(row.fields as Record<string, unknown[]>)[columnId]
+  ) {
+    const innerColumns = Object.fromEntries(
+      Object.entries(row.fields as Record<string, unknown[]>).filter(([key]) => {
+        return key.indexOf(`${columnId}.`) === 0;
+      })
+    );
+    if (isDetails) {
+      // nicely formatted JSON for the expanded view
+      return <span>{JSON.stringify(innerColumns, null, 2)}</span>;
+    }
+
+    // Put the most important fields first
+    const highlights: Record<string, unknown> = (row.highlight as Record<string, unknown>) ?? {};
+    const highlightPairs: Array<[string, string]> = [];
+    const sourcePairs: Array<[string, string]> = [];
+    Object.entries(innerColumns).forEach(([key, values]) => {
+      const subField = indexPattern.getFieldByName(key);
+      const formatter = subField
+        ? indexPattern.getFormatterForField(subField)
+        : { convert: (v: string, ...rest: unknown[]) => v.toString() };
```

Comment:
Even if I can't think of a situation where `null` comes around, `String(v)` would be prepared for this impossible event.